### DISCUSSION TITLE
Testing problems

### DISCUSSION THREAD

**Discussion by josiahgallen** [react-hook-form/react-hook-form#2980] [Q&A] [answered]
I am trying to test my form. When rendering the test with render from `@testing-library/react` I am getting the following error
`TypeError: Object.entries(...).map(...).flat is not a function`
It looks like this error gets thrown from the getPath util in `react-hook-form/src/utils/getPath.ts:15:6`
I have seen other threads that mention upgrading jest to the latest version, I have done this and it does not help.

Started on jest:  26.1.0
Updated to jest: 26.4.2

no improvement was seen.
Please any guidance towards getting this to play nicely with tests.

**Comment by bluebill1049** [comment, marked as answer]
will this section helps? https://react-hook-form.com/advanced-usage#TestingForm

**Reply by josiahgallen** [reply]
I have tried to emulate that and unfortunately am having no luck.

**Reply by josiahgallen** [reply]
Sorry, I forgot, my initial question is answered, I needed to use a newer version of node.
New problem is that `handleSubmit(onSubmit)` mocks are not being fired when simulating submission events. I've seen a handful of threads discussing this and have yet to find the right solution.

**Reply by josiahgallen** [reply]
ending this thread, was passing incorrect schema validation in test setup wrapper which was preventing onSubmit from firing.. just took some time to narrow that down

**Reply by bluebill1049** [reply]
sorry for the late rely... (was busy yesterday) glad you work it out.
